src/main/java/org/apache/sling/resource/collection/impl/ResourceCollectionManagerImpl.java

```diff
@@ -93,14 +92,14 @@ public ResourceCollection createCollection(Resource parentResource, String name,
                 properties = new HashMap<>();
             }
 
-            if (properties.get(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY) != null
-                && !ResourceCollection.RESOURCE_TYPE.equals(properties.get(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY))) {
+            if (properties.get(ResourceResolver.PROPERTY_RESOURCE_TYPE) != null
+                && !ResourceCollection.RESOURCE_TYPE.equals(properties.get(ResourceResolver.PROPERTY_RESOURCE_TYPE))) {
                 properties.put(
-                    JcrResourceConstants.SLING_RESOURCE_SUPER_TYPE_PROPERTY,
+                    "sling:resourceSuperType",
```

Comment:
This property seems to be available as a constant, see https://sling.apache.org/apidocs/sling11/org/apache/sling/api/SlingConstants.html#PROPERTY_RESOURCE_SUPER_TYPE . And I think you use this for resource type as well instead of the ResourceResolver property, for the sake of being consistent.

Reply:
@rombert  great - somehow missed the SlingConstants - thx for pointing out